### PR TITLE
[Lake] Support create table with new parser

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|8|9|88.89%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java [hunk 1]
```diff
@@ -235,7 +240,7 @@ public static void analyze(CreateTableStmt statement, ConnectContext context) {
             rowLengthBytes += columnDef.getType().getStorageLayoutBytes();
         }
 
-        if (rowLengthBytes > Config.max_layout_length_per_row && engineName.equals(DEFAULT_ENGINE_NAME)) {
+        if (rowLengthBytes > Config.max_layout_length_per_row && statement.isOlapOrLakeEngine()) {
             throw new SemanticException("The size of a row (%d) exceed the maximal row size: %d", rowLengthBytes,
                     Config.max_layout_length_per_row);
         }
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:246]: I think this judge can be removed now

**Review by imay — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_be_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:246]: ok, i will remove it in another pr.

**Pull request merged**